src/main/java/com/itimpulse/urlshortener/controller/UrlShortenerController.java

```diff
@@ -18,152 +16,163 @@
 import io.swagger.v3.oas.annotations.responses.ApiResponses;
 import io.swagger.v3.oas.annotations.tags.Tag;
 // import io.swagger.v3.oas.annotations.parameters.RequestBody;
-import org.springframework.web.bind.annotation.RequestBody;
-
+import jakarta.validation.Valid;
 import java.net.URI;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.bind.annotation.RequestBody;
 
 /**
  * REST Controller for URL shortening operations.
- * 
- * This controller provides endpoints for:
- * - Creating shortened URLs with optional TTL and custom IDs
- * - Redirecting to original URLs via shortened IDs
- * - Deleting shortened URLs
- * 
- * All API endpoints follow RESTful conventions and return standardized responses
- * using the CustomResponse wrapper class.
- * 
+ *
+ * <p>This controller provides endpoints for: - Creating shortened URLs with optional TTL and custom
+ * IDs - Redirecting to original URLs via shortened IDs - Deleting shortened URLs
+ *
+ * <p>All API endpoints follow RESTful conventions and return standardized responses using the
+ * CustomResponse wrapper class.
  */
 @RestController
 @RequestMapping
```

Comment:
set `@RequestMapping(value = "/api/v1/url-shortener")`, it will allow to have clean urls like below.

![image](https://github.com/user-attachments/assets/0c4e12a9-8072-4a1f-a411-a257b1f2f4b3)